src/config/config.cpp

```
@@ -540,3 +542,105 @@ void print_default_env() {
+    if (i + 1 < hosts.size()) f << ",";
+    f << "\n";
+  }
+  f << "]\n";
+  return true;
+}
+
+/// Lookup a host entry by IP/hostname, returns "name (note)" if found.
+std::string host_display_name(const std::string& host_port) {
+  const auto& named = Config::instance().named_hosts;
+  auto colon = host_port.rfind(':');
+  std::string h = (colon != std::string::npos) ? host_port.substr(0, colon) : host_port;
```

Clang (14.0.6)

[warning] variable 'h' is not initialized

(cppcoreguidelines-init-variables)